roboflow/__init__.py

```diff
@@ -282,7 +275,7 @@ def project(self, project_name, the_workspace=None):
 
         dataset_info = dataset_info.json()["project"]
 
-        return Project(self.api_key, dataset_info)
+        return Project(self.api_key or "", dataset_info)
```

Comment:
I don't think this is the fix. `api_key` could be `None` for public projects.

I prefer to ignore the error instead of fix the type and cover a possible bug.